src/main/java/httpserver/httpmessages/HTTPResponseParser.java

```diff
@@ -0,0 +1,49 @@
+package httpserver.httpmessages;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+public class HTTPResponseParser {
+
+    private static final String PROTOCOL_VERSION = "HTTP/1.1";
+    private static final String CONTENT_TYPE = "Content-Type : ";
+    private static final String SPACE = " ";
+    private final ByteArrayOutputStream byteArrayOutputStream;
+
+    public HTTPResponseParser(ByteArrayOutputStream byteArrayOutputStream) {
+        this.byteArrayOutputStream = byteArrayOutputStream;
+    }
+
+    public byte[] parse(HTTPResponse httpResponse) {
+        addHeader(httpResponse, byteArrayOutputStream);
+        if (httpResponse.hasBody()) {
+            addBody(httpResponse, byteArrayOutputStream);
+        }
+        return byteArrayOutputStream.toByteArray();
+    }
+
+    private void addBody(HTTPResponse httpResponse, ByteArrayOutputStream byteArrayOutputStream) {
+        try {
+            byteArrayOutputStream.write(getContentType(httpResponse));
+            byteArrayOutputStream.write(httpResponse.getBody());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void addHeader(HTTPResponse httpResponse, ByteArrayOutputStream byteArrayOutputStream) {
```

Comment:
Output arguments like `byteArrayOutputStream` are confusing to me... 
If you think of `methods` to be like questions like `Given 10 and 12, what is the sum?` for `sum(10,12)` then output arguments make little sense: `Given this HTTPresponse and this stream, what is...?` (nothing because there is no return type)